src/main/java/com/google/plantasy/utils/QuizTimingPropertiesUtils.java

```diff
@@ -71,14 +71,22 @@ public static Long getQuizTimestampProperty(String entity, String id_of_entity,
         return null;
     }
 
-    //This function checks if the user has taken the quiz yet by comparing their timestamp with the quiz's timestamp
-    public static boolean userTookQuiz(Long usersQuizTime, Long currentQuizTime) {
+    //This function checks if the user has taken the quiz yet by comparing their timestamp with today's timestamp
+    public static boolean userTookQuiz(Long usersQuizTime) {
         SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd");
         sdf.setTimeZone(TimeZone.getTimeZone("UTC"));
-        String users_quiz_time = sdf.format(usersQuizTime);
-        String current_quiz_time = sdf.format(currentQuizTime);
+        String users_quiz_time;
+        String today_time;
 
-        return(users_quiz_time.compareTo(current_quiz_time) > 0);
+        try {
+            users_quiz_time = sdf.format(usersQuizTime);
+            today_time = sdf.format(new Date());
```

Comment:
we want to make today_time outside of the try catch block. There is no reason for this to fail so we don't need to try catch it.

Reply:
Got it!